rego-templates/trivy-vulns-slack.rego

```diff
@@ -0,0 +1,162 @@
+
+# render_sections split collection of cells provided to chunks of 5 rows each and wraps every chunk with section element
+render_sections(rows, caption, headers) = a {
+	count(rows) > 0 # only if some vulnerabilities are found
+	rows_and_header := array.concat(headers, rows)
+	a := flat_array([s |
+		# code below converts 2 dimension array like [[row1, row2, ... row5], ....]
+		group_size := 10 #it's 5 but every row is represented by 2 items
+		num_chunks := ceil(count(rows) / group_size) - 1
+		indices := {b | b := numbers.range(0, num_chunks)[_] * group_size}
+		some k
+		fields := [array.slice(rows, i, i + group_size) | i := indices[_]][k]
+		# builds markdown section based on slice
+
+    	s := with_caption(fields, caption, k)
+	])
+
+}
+
+render_sections(rows, caption, headers) = [] { #do not render section if provided collection is empty
+	count(rows) == 0
+}
+
+with_caption(fields, caption, position) = s {
+    position == 0
+	s := [
+		{
+			"type": "section",
+			"text": {
+				"type": "mrkdwn",
+				"text": caption,
+			},
+		},
+		{
+			"type": "section",
+			"fields": fields,
+		},
+	]
+}
+with_caption(fields, caption, position) = s {
+    position > 0
+	s := [
+		{
+			"type": "section",
+			"fields": fields,
+		},
+	]
+}
+
+
+###########################################################################################################
+
+vln_list(severity) = l {
+	# builds list of rows for section for the given severity
+	vlnrb := [r |
+		some i, j
+		item := input.Results[i]
+		vlnname := item.Vulnerabilities[j].VulnerabilityID
+
+		fxvrsn := with_default(item.Vulnerabilities[j], "FixedVersion", "none")
+		resource_name = with_default(item.Vulnerabilities[j], "PkgName", "none")
+		resource_version = with_default(item.Vulnerabilities[j], "InstalledVersion", "none")
+
+		item.Vulnerabilities[j].Severity == severity
+
+		r := [
+			{"type": "mrkdwn", "text": vlnname},
+			{"type": "mrkdwn", "text": concat(" / ", [resource_name, resource_version, fxvrsn])},
+		]
+	]
+
+	caption := sprintf("*%s severity vulnerabilities*", [severity]) #TODO make first char uppercase
+
+	headers := [
+		{"type": "mrkdwn", "text": "*Vulnerability ID*"},
+		{"type": "mrkdwn", "text": "*Resource name / Installed version / Fix version*"},
+	]
+
+	# split rows and wrap slices with markdown section
+	l := render_sections(flat_array(vlnrb), caption, headers)
+}
+
+cnt_by_severity(severity) = cnt {
+	vln_list := [r |
+		some i, j
+		item := input.Results[i]
+
+		item.Vulnerabilities[j].Severity == severity
+
+		r := item.Vulnerabilities[j]
+	]
+
+	cnt := count(vln_list)
+}
+
+###########################################################################################################
+postee := with_default(input, "postee", {})
+
+aqua_server := with_default(postee, "AquaServer", "")
+
+title = sprintf("Vulnerability scan report", []) # title is 
+
+aggregation_pkg := "postee.vuls.slack.trivy.aggregation"
```

Comment:
is this used?

Reply:
yeah! It's by default this fields, it's used in the others templates, `title := input.image`

Reply:
I understand, did you forget to check in the aggregate package then? Like this https://github.com/aquasecurity/postee/blob/f7fdfa7378778ace86a7a31f5f5e4b1ca4b8a404/rego-templates/vuls-slack-aggregation.rego#L1-L28

Reply:
I don't see a `postee.vuls.slack.trivy.aggregation` rego package (file).

Reply:
ping @krol3?

Reply:
ok! make sense! I will added